src/TraceEvent/ETWReloggerTraceEventSource.cs

```diff
@@ -157,7 +157,14 @@ public unsafe void WriteEvent(Guid providerId, ref _EVENT_DESCRIPTOR eventDescri
                 newEvent.SetEventDescriptor(ref *ptrDescr);
                 newEvent.SetProviderId(ref providerId);
                 _LARGE_INTEGER fileTimeStamp = new _LARGE_INTEGER();
-                fileTimeStamp.QuadPart = timeStamp.ToFileTimeUtc();
+                if (DateTimeKind.Utc != timeStamp.Kind)
+                {
+                    fileTimeStamp.QuadPart = UTCDateTimeToQPC(timeStamp.ToUniversalTime());
+                }
+                else
+                {
+                    fileTimeStamp.QuadPart = UTCDateTimeToQPC(timeStamp);
```

Comment:
Since this is no longer a file timestamp, will you please rename `fileTimeStamp` to `qpcTimeStamp`?